packages/adapter-drizzle/src/sqlite/index.ts

```diff
@@ -0,0 +1,223 @@
+import { integer, sqliteTable, text, primaryKey } from "drizzle-orm/sqlite-core"
+import { drizzle } from "drizzle-orm/better-sqlite3"
+import crypto from 'node:crypto'
+import Database from "better-sqlite3"
+import { Adapter, AdapterAccount } from "@auth/core/adapters"
+import { eq, and } from "drizzle-orm"
+
+const sqlite = new Database("db.sqlite")
+
+export const users = sqliteTable("users", {
+  id: text("id").notNull().primaryKey(),
+  name: text("name"),
+  email: text("email").notNull(),
+  emailVerified: integer("emailVerified", { mode: "timestamp_ms" }),
+  image: text("image"),
+})
+
+export const accounts = sqliteTable(
+  "accounts",
+  {
+    userId: text("userId")
+      .notNull()
+      .references(() => users.id, { onDelete: "cascade" }),
+    type: text("type").$type<AdapterAccount["type"]>().notNull(),
+    provider: text("provider").notNull(),
+    providerAccountId: text("providerAccountId").notNull(),
+    refresh_token: text("refresh_token"),
+    access_token: text("access_token"),
+    expires_at: integer("expires_at"),
+    token_type: text("token_type"),
+    scope: text("scope"),
+    id_token: text("id_token"),
+    session_state: text("session_state"),
+  },
+  (account) => ({
+    compoundKey: primaryKey(account.provider, account.providerAccountId),
+  })
```

Comment:
Indexes should be made for this table, likely on the following columns:
- `userId` -> uniqueIndex
- `type` -> index
- `provider` -> index
- `expires_at` -> index (descending?) 
- `session_state` -> index

Reply:
https://orm.drizzle.team/docs/indexes-constraints#indexes

Reply:
The adapter's access pattern is only user.id and user.email. For `accounts` it's only by `provider and providerAccountId`, so we don't need indexes for those fields unless you're doing something specific out of scope of the Adapter.

Reply:
Perhaps it would be nice to give the option of accessing these?

Reply:
Users will be able to provide custom tables; you'll be able to add your own custom indices.

Do we think any of these suggestions should be there by default? Going to mark as resolved because it doesn't look like like there should be at a glance?

Reply:
@anthonyshew I think at the very least the indexes should be mentioned in the doc. When you scale to millions of users, this is a big deal.

Reply:
@arivera-xealth We need to ensure we're careful about scope here. The docs need to describe what makes sense for the default configuration. If users are creating extra access patterns such that indices make sense for their application, that doesn't seem like something we need to document in next-auth.

For the default usage of next-auth, do we need to add any extra indices? From what I can tell looking at prior art (namely, the Prisma adapter), it doesn't look like we need to.

Reply:
My 2 cents. I would follow previous implementations on other adapters. Add just the indices those adapters added